### PR TITLE
Upgrade to use the v3 codecov github action for github workflows.

### LINKED ISSUE #4551: Codecov GitHub Action upgrade from V1 to v3
As of February 1, 2022, v1  for Codecov GitHub Action has been fully sunset and no longer functions.
Due to the [deprecation](https://about.codecov.io/blog/introducing-codecovs-new-uploader/) of the underlying bash uploader, the Codecov GitHub Action has released v2/v3 which will use the new [uploader](https://github.com/codecov/uploader).

Upgrade the workflows to use v3.

### PR DESCRIPTION
**Pull request by mgoelswirlds**
**Description**:
Codecov bash uploader has been deprecated starting February 2022. We need to upgrade our workflows to use the codecov v3 GitHub Actions.

* Update workflows to use `codecov/codecov-action@v3`

**Related issue(s)**:

Fixes #4551

**Notes for reviewer**:
Check for all code coverage jobs to pass.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/workflows/web3.yml [hunk 1]
```diff
@@ -37,5 +37,8 @@ jobs:
         working-directory: .
 
       - name: Upload coverage report
+        uses: codecov/codecov-action@v3
         if: always()
         run: bash <(curl -s https://codecov.io/bash)
+        with:
+          fail_ci_if_error: true
```
[the remark this answers is not in the document]
Comment (mgoelswirlds) [reply .github/workflows/web3.yml:44]: So we don't want our builds to fail if codecoverage fails?

**Review by xin-hedera — DISMISSED** [review]
LGTM

File: .github/workflows/grpc-api.yml [hunk 2]
```diff
@@ -37,8 +37,7 @@ jobs:
         working-directory: .
 
       - name: Upload coverage report
-        if: always()
```
Comment (steven-sheehy) [inline .github/workflows/grpc-api.yml:40]: Re-add the `if: always()`. Without this, your coverage report will be greatly under-reported and inaccurate if you have one failing test.

Comment (mgoelswirlds) [reply .github/workflows/grpc-api.yml:40]: Ohh ok, Sorry about that.

**Review by xin-hedera — APPROVED** [review]

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**